python/ray/autoscaler/_private/monitor.py

```diff
@@ -126,6 +132,16 @@ def __init__(self,
         self.autoscaling_config = autoscaling_config
         self.autoscaler = None
 
+        self.prom_metrics = AutoscalerPrometheusMetrics()
+        try:
```

Comment:
Would it be ok to skip this try-except block if `monitor_ip` is none (indicating we're not collecting autoscaler metrics)?
In general, it would be ideal if things looked the same externally as before when `monitor_ip=None`.

The Kubernetes Operator currently runs multiple monitor processes at the same ip, which is why we're skipping doing this support for k8s right now.